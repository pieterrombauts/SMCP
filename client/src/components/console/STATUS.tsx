import React, { useState } from 'react';
import styled from 'styled-components';
import 'bootstrap/dist/css/bootstrap.min.css';
import Form from 'react-bootstrap/Form';
import Button from 'react-bootstrap/Button';
import socket from '../Socket';
import { connect, ConnectedProps } from 'react-redux';
import { RootState } from '../../reducers';

interface AppProps {
  className?: string;
  lobbyID: string;
  userRole: string;
  closeFunction: () => void;
};

const mapState = (state: RootState ) => ({
  time: state.dataReducer.time,
})

const connector = connect(mapState)
type PropsFromRedux = ConnectedProps<typeof connector>
type Props = PropsFromRedux & AppProps

const USTATUS: React.FC<Props> = ( props ) => {
  const subjectRef = React.useRef<any>(null);
  const contentRef = React.useRef<any>(null);
  function sendStatusReport(e: React.FormEvent) {
    e.preventDefault();
    console.log(subjectRef.current.value, contentRef.current.value)
    var newValues = {subject: subjectRef.current.value, content: contentRef.current.value, time: ""};
    console.log(newValues);
    newValues["time"] = props.time;
    socket.emit("STATUS_REPORT", newValues, props.lobbyID);
    subjectRef.current.value = "";
    contentRef.current.value = "";
    props.closeFunction();
  }
  return(
    <Form onSubmit={sendStatusReport}>
      <Form.Group controlId="EFN.Subject">
        <Form.Label> Subject </Form.Label>
        <Form.Control type="text" name="subject" placeholder="Brief description of the status report" ref={subjectRef}/>
      </Form.Group>
      <Form.Group controlId="EFN.Body">
        <Form.Label>Status Report</Form.Label>
        <Form.Control as="textarea" name="content" rows="5" ref={contentRef} />
      </Form.Group>
      <Button variant="primary" type="submit">Submit EFN</Button>
    </Form>
  );
}

const ConSTATUS = styled(USTATUS)``;

export const STATUS = connector(ConSTATUS);