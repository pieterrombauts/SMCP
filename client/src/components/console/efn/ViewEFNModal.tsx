import React from 'react';
import styled from 'styled-components';
import 'bootstrap/dist/css/bootstrap.min.css';
import { EFNDetails } from './EFNDetails'
import { statusReport } from '../customTypes';

interface AppProps {
  className?: string;
  show: boolean;
  closeFunction: () => void;
  reports: statusReport[];
  selectedEfn: string;
  lobbyID: string;
};

export const UViewEFNModal: React.FC<AppProps> = ( props ) => {
  function handleOverlayClick(event: React.MouseEvent<HTMLDivElement>) {
    if (event.target === event.currentTarget) {
      props.closeFunction();
    }
  }
  return(
    <div id="view-efn-modal" className={props.className}>
      <div id="modal-overlay-view-efn" onMouseUp={handleOverlayClick}>
        <div id="modal-content-view-efn">
          <EFNDetails lobbyID={props.lobbyID} reports={props.reports} efnID={props.selectedEfn} closeFunction={props.closeFunction}/>
        </div>
      </div>
    </div>
  );
}

export const ViewEFNModal = styled(UViewEFNModal)`
  visibility: hidden;
  #modal-overlay-view-efn {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    z-index: 9;
    background-color: rgba(0,0,0,0.5);
  }

  #modal-content-view-efn {
    background-color: #FFF;
    position: absolute;
    width: 40%;
    left: 50%;
    top: 50%;
    transform: translate(-50%, -50%);
    border-radius: 10px;
    padding: 40px;
    z-index: 10;
  }
`;