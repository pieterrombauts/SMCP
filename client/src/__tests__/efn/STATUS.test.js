import React from "react";
import 'jest-styled-components'
import toJson from 'enzyme-to-json'
import { ConSTATUS } from '../../components/console/efn/STATUS'
import { shallow, mount, render} from 'enzyme';

describe("EFN form", () => {
  it('renders correctly', () => {
    const wrapper = mount(<ConSTATUS lobbyID={'111111'} userRole={'spartan'} closeFunction={jest.fn()}/>)
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  })
})