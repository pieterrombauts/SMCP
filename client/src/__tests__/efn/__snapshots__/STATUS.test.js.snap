// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EFN form renders correctly 1`] = `
<Styled(USTATUS)
  closeFunction={[MockFunction]}
  lobbyID="111111"
  userRole="spartan"
>
  <USTATUS
    className=""
    closeFunction={[MockFunction]}
    lobbyID="111111"
    userRole="spartan"
  >
    <Form
      inline={false}
      onSubmit={[Function]}
    >
      <form
        className=""
        onSubmit={[Function]}
      >
        <FormGroup
          controlId="EFN.Subject"
        >
          <div
            className="form-group"
          >
            <FormLabel
              column={false}
              srOnly={false}
            >
              <label
                className="form-label"
                htmlFor="EFN.Subject"
              >
                 Subject 
              </label>
            </FormLabel>
            <FormControl
              name="subject"
              placeholder="Brief description of the status report"
              type="text"
            >
              <input
                className="form-control"
                id="EFN.Subject"
                name="subject"
                placeholder="Brief description of the status report"
                type="text"
              />
            </FormControl>
          </div>
        </FormGroup>
        <FormGroup
          controlId="EFN.Body"
        >
          <div
            className="form-group"
          >
            <FormLabel
              column={false}
              srOnly={false}
            >
              <label
                className="form-label"
                htmlFor="EFN.Body"
              >
                Status Report
              </label>
            </FormLabel>
            <FormControl
              as="textarea"
              name="content"
              rows="5"
            >
              <textarea
                className="form-control"
                id="EFN.Body"
                name="content"
                rows="5"
              />
            </FormControl>
          </div>
        </FormGroup>
        <Button
          active={false}
          disabled={false}
          type="submit"
          variant="primary"
        >
          <button
            className="btn btn-primary"
            disabled={false}
            type="submit"
          >
            Submit EFN
          </button>
        </Button>
      </form>
    </Form>
  </USTATUS>
</Styled(USTATUS)>
`;
